Add 4.4-dev5 article (#959)
godotengine/godot-website: collections/_article/dev-snapshot-godot-4-4-dev-5.md

---
title: "Dev snapshot: Godot 4.4 dev 5"

> excerpt: "The PR train stops for no-one!"

Reviewer: I'm not sure "pre-release" is the first thing that comes to mind with "PR" (I think of Pull Request, or Public Relations). Maybe "The pre-release train stops for no-one!" or simpler "The dev train stops for no-one!"?

categories: ["pre-release"]
author: Thaddeus Crews
image: /storage/blog/covers/dev-snapshot-godot-4-4-dev-4.webp
image_caption_title: "Unrailed 2: Back on Track"

> image_caption_description: "A game by Indoor Astronaut"

Reviewer 2: We need to update the title and the image (with the caption)

date: 2024-11-21 12:00:00
---

> With [GodotCon](https://godotengine.org/article/review-godotcon24/) behind us & our developers recuperated, we're thrilled to return

Reviewer: ```suggestion
With [GodotCon](https://godotengine.org/article/review-godotcon24/) behind us and our developers recuperated, we're thrilled to return
```

> to our more frequent release-cycle. It's only been two weeks since we last checked in, yet there's still so much to talk about!

Reviewer: ```suggestion
to a more frequent release-cycle. It's only been two weeks since we last checked in, yet there's still so much to talk about!
```

> So buckle on in for a breakdown of what you can expect to see in dev5.

Reviewer: ```suggestion
So buckle on in for a breakdown of what you can expect to see in 4.4 dev 5.
```

Many of the changes in this release are bug fixes that will be backported to Godot 4.3 and released in 4.3.1! So please
test this release well so we can be confident with the changes and release 4.3.1 with them as soon as possible.

Keep in mind that while we try to make sure each dev snapshot is stable enough for general testing, this is by
definition a pre-release piece of software. Be sure to make frequent backups, or use a version control system such as
Git, to preserve your projects in case of corruption or data loss.

[Jump to the **Downloads** section](#downloads), and give it a spin right now, or continue reading to learn more about improvements in this release. You can also try the [**Web editor**](https://editor.godotengine.org/releases/4.4.dev5/), [**XR editor**](https://www.meta.com/experiences/godot-game-engine/7713660705416473/), or the **Android editor** for this release (join the [Android editor testing group](https://groups.google.com/g/godot-testers) to get access to pre-release builds).

-----

> *The cover illustration is from* [**Unrailed 2: Back on Track**](https://store.steampowered.com/app/2211170/Unrailed_2_Back_on_Track/), *a chaotic co-op railroad construction game where you build your track across procedurally generated worlds! It is developed by [Indoor Astronaut](https://indoorastronaut.ch/). You can purchase the game in early access [on Steam](https://store.steampowered.com/app/2211170/Unrailed_2_Back_on_Track/) and follow the developer on [Twitter](https://x.com/IndoorAstronaut).*

Reviewer 2: We need to update this part as it was dev4 cover.

## Highlights

In case you missed them, see the [4.4 dev 1](/article/dev-snapshot-godot-4-4-dev-1/), [dev2](/article/dev-snapshot-godot-4-4-dev-2/),

> [dev 3](/article/dev-snapshot-godot-4-4-dev-3/), and [dev4](/article/dev-snapshot-godot-4-4-dev-4/) release notes for an overview of

Reviewer: ```suggestion
In case you missed them, see the [4.4 dev 1](/article/dev-snapshot-godot-4-4-dev-1/), [dev 2](/article/dev-snapshot-godot-4-4-dev-2/),
[dev 3](/article/dev-snapshot-godot-4-4-dev-3/), and [dev 4](/article/dev-snapshot-godot-4-4-dev-4/) release notes for an overview of
```

some key features which were already in that snapshot, and are therefore still available for testing in dev 5.

> Here are a few highlights that you might find particularly exciting!

Reviewer: ```suggestion
Here are highlights of a few new features in dev 5 that you might find particularly exciting!
```
(Otherwise it sounded like the below was highlights from dev 1, 2, 3 and 4.)

### Universalize UID support

> Previously, the UID format was limited to only Resource files. This proved to be a pain point for users wishing to reference their scripts and other resources in a manner that's path-agnostic. This, along with a need refactor their scripts anytime these kinds of files were moved, put a significant duty of care on the end-user that shouldn't have been necessary.

Reviewer 2: ```suggestion
Previously, the <abbr title="Universal ID">UID</abbr> format was limited to only Resource files. This proved to be a pain point for users wishing to reference their scripts and other resources in a manner that's path-agnostic. This, along with a need refactor their scripts anytime these kinds of files were moved, put a significant duty of care on the end-user that shouldn't have been necessary.
```

> Previously, the <abbr title="Universal ID">UID</abbr> format was limited to only Resource files. This proved to be a pain point for users wishing to reference their scripts and other resources in a manner that's path-agnostic. This, along with a need refactor their scripts anytime these kinds of files were moved, put a significant duty of care on the end-user that shouldn't have been necessary.

Reviewer: ```suggestion
Previously, the <abbr title="Universal ID">UID</abbr> format was limited to only Resource files. This proved to be a pain point for users wishing to reference their scripts and other resources in a manner that's path-agnostic. This, along with a need to refactor their scripts anytime these kinds of files were moved, put a significant duty of care on the end-user that shouldn't have been necessary.
```

> Starting with dev5, this will no longer be your burden to bear! Thanks to a long-term effort from [reduz](https://github.com/reduz), UIDs will now be applied universally in a way the engine can automatically track and account for. This is achived via `.uid` files for the resource types that previously didn't support them, functioning similarly to other metadata files that "track" a main file.

Reviewer: ```suggestion
Starting with dev 5, this will no longer be your burden to bear! Thanks to a long-term effort from [reduz](https://github.com/reduz), UIDs will now be applied universally in a way the engine can automatically track and account for. This is achieved via `.uid` files for the resource types that previously didn't support them, functioning similarly to other metadata files that "track" a main file.
```

> (Note that, unlike metadata files, `.uid` files are strictly for the editor; the information is migrated to the uid database on export. For more information, see ([GH-97352](https://github.com/godotengine/godot/pull/97352)).)

Reviewer 2: ```suggestion
(Note that, unlike metadata files, `.uid` files are strictly for the editor; the information is migrated to the uid database on export. For more information, see [GH-97352](https://github.com/godotengine/godot/pull/97352).)
```

Reviewer 2: Maybe we could tell users to add those files to their versioning software. We tell them that's editor only, but the file still needs to be synced among the team.

Author: The syntax for this is actually correct, based on the past articles; I had *everything else* wrong, whoops!

Also good call, added a warning regarding version control

> Note that, unlike metadata files, `.uid` files are strictly for the editor; the information is migrated to the uid database on export. Despite this, users using version contorl software **should add these files**, as they're required to properly sync data. For more information, see ([GH-97352](https://github.com/godotengine/godot/pull/97352)).

Reviewer: ```suggestion
Note that, unlike metadata files, `.uid` files are strictly for the editor; the information is migrated to the uid database on export. Despite this, users using version control software **should add these files**, as they're required to properly sync data. For more information, see ([GH-97352](https://github.com/godotengine/godot/pull/97352)).
```

### Favorite editor items

> A common complaint we hear regarding the viewport is the potential for it to get cluttered. Namely, while there's generally a wide selection of options available for a given class/script, users will usually only care about a particular subsection that suits them. To account for this, [YeldhamDev](https://github.com/Hilderin) brings us the long-awaited ability to pin one's favorite properties in the inspector! Check out the implementation from PR [GH-97352](https://github.com/godotengine/godot/pull/97415) below:

Reviewer 2: ~~Either~~ it's YeldhamDev ~~or Hilderin~~! 😆 

```suggestion
A common complaint we hear regarding the viewport is the potential for it to get cluttered. Namely, while there's generally a wide selection of options available for a given class/script, users will usually only care about a particular subsection that suits them. To account for this, [YeldhamDev](https://github.com/YeldhamDev) brings us the long-awaited ability to pin one's favorite properties in the inspector! Check out the implementation from PR [GH-97352](https://github.com/godotengine/godot/pull/97415) below:
```

<video autoplay loop muted playsinline>
  <source src="/storage/blog/dev-snapshot-godot-4-4-dev-5/favorite-inspector.mp4?1" type="video/mp4">
</video>

### And more!

There are too many exciting changes to list them all here, but here's a curated selection:

- 2D: Add a way to know when a TileMapLayer's cell is modified ([GH-96188](https://github.com/godotengine/godot/pull/96188)).
- 2D: Make possible to scale multiple nodes at once in the canvas editor ([GH-98534](https://github.com/godotengine/godot/pull/98534)).

> - 3D: Fix Gridmap shortcut conflicts with 3d editor ([GH-99170](https://github.com/godotengine/godot/pull/99170)).

Reviewer: ```suggestion
- 3D: Fix Gridmap shortcut conflicts with 3D editor ([GH-99170](https://github.com/godotengine/godot/pull/99170)).
```
